### PR TITLE
Add test to verify failed jobs are not used as cache hits

### PR DESCRIPTION
**Pull request by Horneth**

### CODE REVIEW

File: src/main/resources/standardTestCases/failing_continue_on_return_code.test [hunk 1]
```diff
@@ -4,3 +4,7 @@ testFormat: workflowfailure
 files {
   wdl: failing_continue_on_return_code/failing_continue_on_return_code.wdl
 }
+
+metadata {
+    "calls.w.A.callCaching.allowResultReuse": false
+}
```
Comment (mcovarr) [inline src/main/resources/standardTestCases/failing_continue_on_return_code.test:10]: \n

File: src/main/resources/standardTestCases/dont_cache_to_failed_jobs/dont_cache_to_failed_jobs.wdl [hunk 2]
```diff
@@ -0,0 +1,16 @@
+task failing_task {
+    command {
+        # C'est nul !
+        exit 1
+    }
+    runtime {
+        docker: "ubuntu@sha256:71cd81252a3563a03ad8daee81047b62ab5d892ebbfbf71cf53415f29c130950"
```
Comment (mcovarr) [inline src/main/resources/standardTestCases/dont_cache_to_failed_jobs/dont_cache_to_failed_jobs.wdl]: is there a reason to go with SHAs now?

**Review by mcovarr — APPROVED** [review]

**Comment by mcovarr** [comment]
:+1:

[Approved with PullApprove](https://pullapprove.com/broadinstitute/centaur/pull-request/203/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=broadinstitute/centaur)

File: src/main/resources/standardTestCases/dont_cache_to_failed_jobs/dont_cache_to_failed_jobs.wdl (hunk 2 above)
Comment (Horneth) [reply src/main/resources/standardTestCases/dont_cache_to_failed_jobs/dont_cache_to_failed_jobs.wdl]: not that I know of, probably just copy pasted from another test

File: src/main/scala/centaur/test/standard/CentaurTestFormat.scala [hunk 3]
```diff
@@ -13,6 +13,7 @@ sealed abstract class CentaurTestFormat(val name: String) {
     case WorkflowFailureTest => "fail during execution"
     case RunTwiceExpectingCallCachingTest => "call cache the second run of"
     case RunTwiceExpectingNoCallCachingTest => "NOT call cache the second run of"
+    case RunFailingTwiceExpectingNoCallCachingTest => "NOT call cache the second run of"
```
Comment (ruchim) [inline src/main/scala/centaur/test/standard/CentaurTestFormat.scala]: "NOT call cache the second failing run of" maybe to differentiate it from the other test spec of RunTwiceExpectingNoCallCachingTest

**Review by ruchim — APPROVED** [review]
👍

File: src/main/scala/centaur/test/standard/CentaurTestFormat.scala (hunk 3 above)
Comment (Horneth) [reply src/main/scala/centaur/test/standard/CentaurTestFormat.scala]: oh yeah good catch :)

**Pull request merged**